src/playwright-api/takeArchive.ts

```diff
@@ -36,12 +46,43 @@ async function takeArchive(
   });
 
   // Serialize and capture the DOM
-  const domSnapshot: serializedNode = await page.evaluate(dedent`
+  const domSnapshot: SerializedNode = await page.evaluate(dedent`
     ${rrweb};
     rrwebSnapshot.snapshot(document, { noAbsolute: true });
   `);
 
+  // XXX_jwir3: We go through and filter any of these that would have file names that would be too large.
+  const sourceMap: Map<string, string> = new Map<string, string>();
+  if (domSnapshot.childNodes.length !== 0) {
+    shortenFileNameSrc(domSnapshot.childNodes, sourceMap);
+  }
+
   testInfo.attach(name, { contentType, body: JSON.stringify(domSnapshot) });
+
+  return Promise.resolve(sourceMap);
+}
+
+function shortenFileNameSrc(
+  input: any,
+  existingSourceMap: Map<string, string>
+): Map<string, string> {
+  for (let i = 0; i < input.length; i += 1) {
+    const nextChildNode = input[i];
```

Comment:
Nitpick: could simplify to

```suggestion
  for (const nextChildNode of input) {
```